src/schema/v2/previewSavedSearch.ts

```diff
@@ -99,3 +109,46 @@ export const PreviewSavedSearchField: GraphQLFieldConfig<
     return args.attributes
   },
 }
+
+const generateSuggestedName = async (
+  parent,
+  _args,
+  { artistLoader },
+  _info
+) => {
+  const suggestedNameElements: string[] = []
+
+  const { artistIDs, priceRange, additionalGeneIDs, attributionClass } = parent
+
+  // artist
+  const artistLabels = (await getArtistLabels(artistIDs, artistLoader)).map(
+    (l) => l.displayValue
+  )
+  if (artistLabels.length) {
+    suggestedNameElements.push(artistLabels.join(" or "))
```

Comment:
Not explicitly mentioned in the ticket but we need to handle array-valued inputs to most of these fields. When multiple values are supplied these act as `OR`'d terms, so the naming reflects this.